### PR TITLE
Fix 266 Add authorized=null option in PropertyFilter

### LINKED ISSUE #266: Property filters should match status name
Currently it show "Yes", "No" and "Unknown". Where we should read, depending on the status field, each statuses names + "All". 

So for instance for the authorization status: "Authorized", "Not Yet authorized", "Not authorized" and "All".

### PR DESCRIPTION
**Pull request by 2ynn**
Fixes #266
----------
## Type of change:
- Bug fix (change which fixes an issue).
----------
## What's Changed:
- changed `authorized` BooleanFilter into ChoiceFilter in PropertyFilter to allow filtering by all three options: null / true / false
- current year is no longer hardcoded in HarvestFilter
- general cleanup in filters
- removed CRSF token in all filter forms (GET request)
----------
## Affected urls:
- localhost:8000/property/?authorized=2
- localhost:8000/property/?authorized=1
- localhost:8000/property/?authorized=0
-------
## Checklist:
- [x] My code follows the [style guidelines](https://github.com/LesFruitsDefendus/saskatoon-ng/blob/develop/CONTRIBUTING.md) of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] New and existing unit tests pass locally with my changes.

### CODE REVIEW

File: saskatoon/sitebase/templates/app/list_views/filters.html [hunk 1]
```diff
@@ -18,7 +18,6 @@ <h4 class="panel-title">
             <div class="collapse" id="accordionGreen-one" role="tabpanel">
                 <div class="panel-body" style="display:inline-block;">
                     <form role="form">
-                        {% csrf_token %}
```
Comment (tristanlatr) [inline saskatoon/sitebase/templates/app/list_views/filters.html:21]: Why this change exactly ?

File: saskatoon/harvest/filters.py [hunk 2]
```diff
@@ -50,51 +61,82 @@ class HarvestFilter(filters.FilterSet):
...
+
+    pending = filters.BooleanFilter(
+        help_text="",
+        label=_("Pending validation")
+    )
+
+    def authorized_filter(self, queryset, name, choice):
+        if choice == '2':
+            return queryset.filter(authorized__isnull=True)
+        return queryset.filter(authorized=bool(int(choice)))
```
Comment (tristanlatr) [inline saskatoon/harvest/filters.py:125]: Would it make send to wrap this logic inside a custom `BooleanOptionalFilter` class? Maybe there are other optional boolean fields that would benefit from using it ?

This is a minor comment and does not have to be implemented in this PR.

**Review by tristanlatr — APPROVED** [review]
Looks great, thank you @2ynn

File: saskatoon/sitebase/templates/app/list_views/filters.html (hunk 1 above)
Comment (2ynn) [reply saskatoon/sitebase/templates/app/list_views/filters.html:21]: As far as I know a crsf token is not needed for a GET request and it's really annoying to have it in the url while testing filters
